platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationImpl.java

```diff
@@ -40,93 +39,34 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(ApplicationImpl.class);
 
-  private Set<Feature> features = new HashSet<>();
-
-  private Set<Feature> autoInstallFeatures = new HashSet<>();
-
-  private Feature mainFeature;
-
   private String name;
 
-  private String version;
-
   private String description;
 
-  private URI location;
-
-  /**
-   * Creates a new instance of application.
-   *
-   * @param repo Creates the application from a Karaf Feature Repository object.
-   */
-  public ApplicationImpl(Repository repo) {
-    String repoName = null;
-    try {
-      repoName = repo.getName();
-      features.addAll(Arrays.asList(repo.getFeatures()));
-    } catch (Exception e) {
-      throw new IllegalStateException(e);
-    }
-    setRepoNameAndVersion(repoName);
-    location = repo.getURI();
-    if (features.size() == 1) {
-      autoInstallFeatures.add(features.iterator().next());
-    } else {
-      autoInstallFeatures.addAll(
-          features
-              .stream()
-              .filter(
-                  curFeature ->
-                      StringUtils.equalsIgnoreCase(
-                          Feature.DEFAULT_INSTALL_MODE, curFeature.getInstall()))
-              .collect(Collectors.toList()));
-    }
-    // Determine mainFeature
-    if (autoInstallFeatures.size() == 1) {
-      mainFeature = autoInstallFeatures.iterator().next();
-      name = mainFeature.getName();
-      version = mainFeature.getVersion();
-      description = mainFeature.getDescription();
-    } else {
-      Optional<Feature> first = features.stream().filter(f -> name.equals(f.getName())).findFirst();
-      if (first.isPresent()) {
-        mainFeature = first.get();
-        name = mainFeature.getName();
-        version = mainFeature.getVersion();
-        description = mainFeature.getDescription();
+  private List<String> bundleLocations;
```

Comment:
Boon is the only one setting this field. I can create a default constructor.

Reply:
just add a comment that it gets initialized via Boon bean mapping

Reply:
By design, it means it could remain `null` if Boon doesn't see this attribute in the Json. You should add protection code against null above in `getBundles()`